scenarios/throughput_stress.go

```diff
@@ -134,8 +138,17 @@ func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error
 					NexusEndpoint:                nexusEndpoint,
 					SleepActivityPerPriority:     sleepActivityPerPriority,
 				})
-			// The 1 is for the final workflow run
-			t.workflowCount.Add(uint64(result.TimesContinued + result.ChildrenSpawned + 1))
+
+			if err == nil {
+				// The 1 is for the final workflow run.
+				curTotal := t.workflowCount.Add(uint64(result.TimesContinued + result.ChildrenSpawned + 1))
+
+				info.StatusCallback(tpsStatusUpdate{
+					CompletedIteration: curIteration,
```

Comment:
The caller will need to make sure to store the "low water mark"; and not just the latest iteration.